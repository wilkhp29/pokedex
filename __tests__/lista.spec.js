import React from 'react';
import { mount } from 'enzyme';
import Lista from '../src/Views/list';

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

const initalState = {
    pokedox:[],
    pokemon:{busca:'',pokemon:{},loading:false,pokemonBusca:{}}
}

const store = mockStore(initalState);

describe('lista', () => {
    it('renderizando a Lista', () => {
        const wrapper = mount(
        <Provider store={store}>
            <Lista match={{params:{name:"fire"}}}/>
        </Provider>
            ,{context:{store}});
    
        expect(wrapper).toMatchSnapshot();
    });
})