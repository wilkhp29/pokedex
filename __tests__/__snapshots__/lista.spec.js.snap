// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lista renderizando a Lista 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <List
    match={
      Object {
        "params": Object {
          "name": "fire",
        },
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#fc6963",
          "borderRadius": 5,
          "color": "#fff",
          "marginRight": 10,
          "padding": 10,
        }
      }
    >
      Voltar
    </button>
    <h1
      style={
        Object {
          "alignItems": "center",
          "background": "#fff",
          "felx": 1,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Carregando lista ...
    </h1>
  </List>
</Provider>
`;
